src/engine/scheduler.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  EngineInterface,
+  SchedulerInterface,
+  FrameInterface
+} from './interfaces';
+import W from './window-proxy';
+
+import { default as Queue, DOMQueue} from './queues/queue';
+
+export class Frame implements FrameInterface {
+  constructor(
+    public timestamp: number,
+    public scrollTop: number,
+    public scrollLeft: number,
+    public width: number,
+    public height: number
+  ) {}
+}
+
+export function uuid() {
```

Comment:
yes, but why is it using this format? this is an internal id? what is it used for? does it need to be a string? can it be an integer?